File: lib/genetic_algorithm/individuals/InputKeyIndividual.py
from lib.genetic_algorithm.individuals.InputIndividual import InputIndividual
from lib.input_scripts.InputScript import InputScript


class InputKeyIndividual(InputIndividual):
    """
    An individual for the genetic algorithm, it represents an input script. Its composed by only single key genes
    """

    def __init__(self, genes):
        """
        :param genes: <list(SingleKeyGene)> The list of genes of the current individual
        """
        InputIndividual.__init__(self, genes)

    def get_inputs(self):
        """
        Gets the input script of the individual
        :return: <InputScript> the input script that this individual represents
        """
        inputs = InputScript()

        for i in range(len(self.genes)):
            gene_input = self.genes[i].get_key_input()
            inputs.add_input(gene_input, i+1, i+2)

        return inputs

File: love_ga_wrapper/love_ga_wrapper/lib/genetic_algorithms/individuals/InputKeyIndividual.lua
local class = require("love_ga_wrapper.lib.classes.class")
local extend = require("love_ga_wrapper.lib.classes.extend")
local InputIndividual = require("love_ga_wrapper.lib.genetic_algorithms.individuals.InputIndividual")
-------------------------------------------------------------

-- class: InputKeyIndividual
-- param: genomes:list(SingleInputKeyGenome) -> The list of genomes of the curren individual
-- An individual in the genetic algorithm
local InputKeyIndividual = extend(InputIndividual, function(self, genomes)

end)

-- getInputsOnFrame: int(1,) -> dict(str,bool)
-- Gets this individual's inputs given a frame
function InputKeyIndividual.getInputsOnFrame(self, frame)
    return self.genomes[frame]:getInputs()
end

-- getInputs: None -> list(dict(str, bool))
-- Gets the complete list of this individual's inputs
function InputKeyIndividual.getInputs(self)
    local inputs = {}
    inputs[1] = {}
    for i = 1, (# self.genomes) do
        local genome_inputs = self.genomes[i]:getInputs()

        inputs[i+1] = {}

        for input, status in pairs(genome_inputs) do
            inputs[i][input] = status
            if status then
                inputs[i+1][input] = false
            end
        end
    end
    return inputs
end

-- getClass: None -> class
-- Returns the class of the Input Key Individual
function InputKeyIndividual.getClass(self)
    return InputKeyIndividual
end

return InputKeyIndividual